Mathlib/Topology/LocalAtTarget.lean

```diff
@@ -65,15 +65,15 @@ alias ClosedEmbedding.restrictPreimage := Set.restrictPreimage_closedEmbedding
 
 theorem Set.restrictPreimage_isClosedMap (s : Set β) (H : IsClosedMap f) :
     IsClosedMap (s.restrictPreimage f) := by
-  rintro t ⟨u, hu, e⟩
-  refine' ⟨⟨_, (H _ (IsOpen.isClosed_compl hu)).1, _⟩⟩
-  rw [← (congr_arg HasCompl.compl e).trans (compl_compl t)]
-  simp only [Set.preimage_compl, compl_inj_iff]
-  ext ⟨x, hx⟩
-  suffices (∃ y, y ∉ u ∧ f y = x) ↔ ∃ y, y ∉ u ∧ f y ∈ s ∧ f y = x by
-    simpa [Set.restrictPreimage, ← Subtype.coe_inj]
-  exact ⟨fun ⟨a, b, c⟩ => ⟨a, b, c.symm ▸ hx, c⟩, fun ⟨a, b, _, c⟩ => ⟨a, b, c⟩⟩
-#align set.restrict_preimage_is_closed_map Set.restrictPreimage_isClosedMap
+  intro t ⟨u, hu, e⟩
+  refine' ⟨(f '' uᶜ)ᶜ, isOpen_compl_iff.2 (H uᶜ (IsOpen.isClosed_compl hu)), _⟩
```

Comment:
Here's a Lean trick: you can simply a tactic proof as a proof term using `by`: in plain English,
you can write "by simp [e, ← image_restrictPreimage]" in place of the last place-holder.

Once you've done that, you have no more goals left, and it's good practice to use `exact instead`.

Reply:
I shortened this proof the same way I shortened the following one.

Reply:
I was on the fence whether only term mode or two lines ("intro bla; exact baz") is better. :-)
If a maintainer has a strong opinion, we'll see!